pytorch_lightning/utilities/data.py

```diff
@@ -47,6 +48,11 @@ def _extract_batch_size(batch: BType) -> Generator[int, None, None]:
         if isinstance(batch, Mapping):
             batch = batch.values()
 
+        for sample in batch:
+            yield from _extract_batch_size(sample)
+    elif is_dataclass(batch):
+        batch = [getattr(batch, field) for field in batch.__dataclass_fields__]
+
         for sample in batch:
```

Comment:
maybe iterate directly over `[getattr(batch, field.name) for field in dataclasses.fields(batch)]` rather than storing it?

Reply:
Then we could also do:
```
for field in dataclasses.fields(batch):
    yield from _extract_batch_size(getattr(batch, field.name))
```
I honestly dont care 😄

Reply:
anyway is fine.. since they are using references :)